test/integration/spec/connect.js

```diff
@@ -146,6 +146,35 @@ describe('connect', function() {
   });
 
   describe('signaling region', async () => {
+    let sid;
+    let token;
+    beforeEach(async () => {
+      const identity = randomName();
+      token = getToken(identity);
+      sid = await createRoom(randomName(), defaults.topology);
+    });
+
+    // few examples of invalid regions.
+    ['invalid', '34', '$abc', 'abc!'].forEach((region) => {
```

Comment:
@makarandp0 Instead of a separate test here, I would include them in the existing test as follows:
```js
const invalidRegions = ['foo', '34', '$abc', 'abc!'];

const validRegions = defaults.regions ? defaults.regions.split(',') : [
  'au1', 'br1', 'de1', 'ie1', 'in1', 'jp1', 'sg1', 'us1', 'us2'
];

const regions = ['without', 'gll', ...invalidRegions, ...validRegions];

regions.forEach(region => {
  const regionOptions = region === 'without' ? {} : { region };
  const scenario = `when called ${region === 'without'
    ? 'without a region'
    : `with a region that is ${invalidRegions.includes(region) ? 'invalid' : 'valid'} (${region})`};

  ...

  it(`should return a CancelablePromise that ${invalidRegions.includes(region)' ? 'rejects with a SignalingConnectionError' : 'resolves with a Room'}`, async () => {...});
});
```